lib/facility.rb

```diff
@@ -1,14 +1,76 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def has_service?(service)
+    @services.include?(service)
+   end
+
+  def allow_service(service)
```

Comment:
I'm not sure this method is necessary since you created the `has_service?` method to check, that method will return a true/false that you can use in conditionals.

Reply:
I agree and have seen other versions of projects that don't include it. Those look cleaner and are more simple. I was overcomplicating it.